### PR TITLE
InfluxDB_database: Make the path argument conditional on the InfluxDB version

### LINKED ISSUE #997: influxdb_database got an unexpected keyword argument 'path'
##### SUMMARY

The influxdb_database module fails to execute on remote hosts with a TypeError.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

influxdb_database

##### ANSIBLE VERSION

```paste below
ansible 2.10.1
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/exploit/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.8/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.8.5 (default, Sep  5 2020, 10:50:12) [GCC 10.2.0]
```

##### CONFIGURATION

```paste below

```

##### OS / ENVIRONMENT

Remote OS is ubuntu 16.04 with both Python2.7 and Python3.5
Ansible is launched from an ArchLinux machine using Python3.8

##### STEPS TO REPRODUCE

```yaml
---

- hosts: all
  gather_facts: no
  tasks:
    - influxdb_database:
        hostname: localhost
        database_name: foobar
        port: 8086
        state: present
```

##### EXPECTED RESULTS

Module should execute and create the required database in the local InfluxDB instance.

Works fine in Ansible 2.9

##### ACTUAL RESULTS

Module fails to execute and raises a `TypeError got an unexpected keyword argument 'path'`.

```paste below
$ ansible-playbook -i 172.20.42.11, mcve.yml -k -vvv
ansible-playbook 2.10.1
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/exploit/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.8/site-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 3.8.5 (default, Sep  5 2020, 10:50:12) [GCC 10.2.0]
Using /etc/ansible/ansible.cfg as config file
SSH password: 
Parsed 172.20.42.11, inventory source with host_list plugin
redirecting (type: modules) ansible.builtin.influxdb_database to community.general.influxdb_database

PLAYBOOK: mcve.yml ****************************************************************************************************************************************************************************************************************************
1 plays in mcve.yml

PLAY [all] ************************************************************************************************************************************************************************************************************************************
META: ran handlers

TASK [influxdb_database] **********************************************************************************************************************************************************************************************************************
task path: /home/exploit/code_testing/ansible/mcve.yml:6
<172.20.42.11> ESTABLISH SSH CONNECTION FOR USER: None
<172.20.42.11> SSH: EXEC sshpass -d10 ssh -C -o ControlMaster=auto -o ControlPersist=60s -o ConnectTimeout=10 -o ControlPath=/home/exploit/.ansible/cp/9bfdd0dc66 172.20.42.11 '/bin/sh -c '"'"'echo ~ && sleep 0'"'"''
<172.20.42.11> (0, b'/home/exploit\n', b'')
<172.20.42.11> ESTABLISH SSH CONNECTION FOR USER: None
<172.20.42.11> SSH: EXEC sshpass -d10 ssh -C -o ControlMaster=auto -o ControlPersist=60s -o ConnectTimeout=10 -o ControlPath=/home/exploit/.ansible/cp/9bfdd0dc66 172.20.42.11 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo /home/exploit/.ansible/tmp `"&& mkdir "` echo /home/exploit/.ansible/tmp/ansible-tmp-1601368297.2341213-9414-127682684873832 `" && echo ansible-tmp-1601368297.2341213-9414-127682684873832="` echo /home/exploit/.ansible/tmp/ansible-tmp-1601368297.2341213-9414-127682684873832 `" ) && sleep 0'"'"''
<172.20.42.11> (0, b'ansible-tmp-1601368297.2341213-9414-127682684873832=/home/exploit/.ansible/tmp/ansible-tmp-1601368297.2341213-9414-127682684873832\n', b'')
redirecting (type: modules) ansible.builtin.influxdb_database to community.general.influxdb_database
<172.20.42.11> Attempting python interpreter discovery
<172.20.42.11> ESTABLISH SSH CONNECTION FOR USER: None
<172.20.42.11> SSH: EXEC sshpass -d10 ssh -C -o ControlMaster=auto -o ControlPersist=60s -o ConnectTimeout=10 -o ControlPath=/home/exploit/.ansible/cp/9bfdd0dc66 172.20.42.11 '/bin/sh -c '"'"'echo PLATFORM; uname; echo FOUND; command -v '"'"'"'"'"'"'"'"'/usr/bin/python'"'"'"'"'"'"'"'"'; command -v '"'"'"'"'"'"'"'"'python3.7'"'"'"'"'"'"'"'"'; command -v '"'"'"'"'"'"'"'"'python3.6'"'"'"'"'"'"'"'"'; command -v '"'"'"'"'"'"'"'"'python3.5'"'"'"'"'"'"'"'"'; command -v '"'"'"'"'"'"'"'"'python2.7'"'"'"'"'"'"'"'"'; command -v '"'"'"'"'"'"'"'"'python2.6'"'"'"'"'"'"'"'"'; command -v '"'"'"'"'"'"'"'"'/usr/libexec/platform-python'"'"'"'"'"'"'"'"'; command -v '"'"'"'"'"'"'"'"'/usr/bin/python3'"'"'"'"'"'"'"'"'; command -v '"'"'"'"'"'"'"'"'python'"'"'"'"'"'"'"'"'; echo ENDFOUND && sleep 0'"'"''
<172.20.42.11> (0, b'PLATFORM\nLinux\nFOUND\n/usr/bin/python\n/usr/bin/python3.5\n/usr/bin/python2.7\n/usr/bin/python3\n/usr/bin/python\nENDFOUND\n', b'')
<172.20.42.11> ESTABLISH SSH CONNECTION FOR USER: None
<172.20.42.11> SSH: EXEC sshpass -d10 ssh -C -o ControlMaster=auto -o ControlPersist=60s -o ConnectTimeout=10 -o ControlPath=/home/exploit/.ansible/cp/9bfdd0dc66 172.20.42.11 '/bin/sh -c '"'"'/usr/bin/python && sleep 0'"'"''
<172.20.42.11> (0, b'{"osrelease_content": "NAME=\\"Ubuntu\\"\\nVERSION=\\"16.04.6 LTS (Xenial Xerus)\\"\\nID=ubuntu\\nID_LIKE=debian\\nPRETTY_NAME=\\"Ubuntu 16.04.6 LTS\\"\\nVERSION_ID=\\"16.04\\"\\nHOME_URL=\\"http://www.ubuntu.com/\\"\\nSUPPORT_URL=\\"http://help.ubuntu.com/\\"\\nBUG_REPORT_URL=\\"http://bugs.launchpad.net/ubuntu/\\"\\nVERSION_CODENAME=xenial\\nUBUNTU_CODENAME=xenial\\n", "platform_dist_result": ["Ubuntu", "16.04", "xenial"]}\n', b'')
Using module file /usr/lib/python3.8/site-packages/ansible_collections/community/general/plugins/modules/influxdb_database.py
<172.20.42.11> PUT /home/exploit/.ansible/tmp/ansible-local-9410j_m2k00u/tmpmv0_dg5o TO /home/exploit/.ansible/tmp/ansible-tmp-1601368297.2341213-9414-127682684873832/AnsiballZ_influxdb_database.py
<172.20.42.11> SSH: EXEC sshpass -d10 sftp -o BatchMode=no -b - -C -o ControlMaster=auto -o ControlPersist=60s -o ConnectTimeout=10 -o ControlPath=/home/exploit/.ansible/cp/9bfdd0dc66 '[172.20.42.11]'
<172.20.42.11> (0, b'sftp> put /home/exploit/.ansible/tmp/ansible-local-9410j_m2k00u/tmpmv0_dg5o /home/exploit/.ansible/tmp/ansible-tmp-1601368297.2341213-9414-127682684873832/AnsiballZ_influxdb_database.py\n', b'')
<172.20.42.11> ESTABLISH SSH CONNECTION FOR USER: None
<172.20.42.11> SSH: EXEC sshpass -d10 ssh -C -o ControlMaster=auto -o ControlPersist=60s -o ConnectTimeout=10 -o ControlPath=/home/exploit/.ansible/cp/9bfdd0dc66 172.20.42.11 '/bin/sh -c '"'"'chmod u+x /home/exploit/.ansible/tmp/ansible-tmp-1601368297.2341213-9414-127682684873832/ /home/exploit/.ansible/tmp/ansible-tmp-1601368297.2341213-9414-127682684873832/AnsiballZ_influxdb_database.py && sleep 0'"'"''
<172.20.42.11> (0, b'', b'')
<172.20.42.11> ESTABLISH SSH CONNECTION FOR USER: None
<172.20.42.11> SSH: EXEC sshpass -d10 ssh -C -o ControlMaster=auto -o ControlPersist=60s -o ConnectTimeout=10 -o ControlPath=/home/exploit/.ansible/cp/9bfdd0dc66 -tt 172.20.42.11 '/bin/sh -c '"'"'/usr/bin/python /home/exploit/.ansible/tmp/ansible-tmp-1601368297.2341213-9414-127682684873832/AnsiballZ_influxdb_database.py && sleep 0'"'"''
<172.20.42.11> (1, b'Traceback (most recent call last):\r\n  File "/home/exploit/.ansible/tmp/ansible-tmp-1601368297.2341213-9414-127682684873832/AnsiballZ_influxdb_database.py", line 102, in <module>\r\n    _ansiballz_main()\r\n  File "/home/exploit/.ansible/tmp/ansible-tmp-1601368297.2341213-9414-127682684873832/AnsiballZ_influxdb_database.py", line 94, in _ansiballz_main\r\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\r\n  File "/home/exploit/.ansible/tmp/ansible-tmp-1601368297.2341213-9414-127682684873832/AnsiballZ_influxdb_database.py", line 40, in invoke_module\r\n    runpy.run_module(mod_name=\'ansible_collections.community.general.plugins.modules.influxdb_database\', init_globals=None, run_name=\'__main__\', alter_sys=True)\r\n  File "/usr/lib/python2.7/runpy.py", line 188, in run_module\r\n    fname, loader, pkg_name)\r\n  File "/usr/lib/python2.7/runpy.py", line 82, in _run_module_code\r\n    mod_name, mod_fname, mod_loader, pkg_name)\r\n  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code\r\n    exec code in run_globals\r\n  File "/tmp/ansible_influxdb_database_payload_k1Cvhm/ansible_influxdb_database_payload.zip/ansible_collections/community/general/plugins/modules/influxdb_database.py", line 141, in <module>\r\n  File "/tmp/ansible_influxdb_database_payload_k1Cvhm/ansible_influxdb_database_payload.zip/ansible_collections/community/general/plugins/modules/influxdb_database.py", line 123, in main\r\n  File "/tmp/ansible_influxdb_database_payload_k1Cvhm/ansible_influxdb_database_payload.zip/ansible_collections/community/general/plugins/module_utils/influxdb.py", line 88, in connect_to_influxdb\r\nTypeError: __init__() got an unexpected keyword argument \'path\'\r\n', b'Shared connection to 172.20.42.11 closed.\r\n')
<172.20.42.11> Failed to connect to the host via ssh: Shared connection to 172.20.42.11 closed.
<172.20.42.11> ESTABLISH SSH CONNECTION FOR USER: None
<172.20.42.11> SSH: EXEC sshpass -d10 ssh -C -o ControlMaster=auto -o ControlPersist=60s -o ConnectTimeout=10 -o ControlPath=/home/exploit/.ansible/cp/9bfdd0dc66 172.20.42.11 '/bin/sh -c '"'"'rm -f -r /home/exploit/.ansible/tmp/ansible-tmp-1601368297.2341213-9414-127682684873832/ > /dev/null 2>&1 && sleep 0'"'"''
<172.20.42.11> (0, b'', b'')
The full traceback is:
Traceback (most recent call last):
  File "/home/exploit/.ansible/tmp/ansible-tmp-1601368297.2341213-9414-127682684873832/AnsiballZ_influxdb_database.py", line 102, in <module>
    _ansiballz_main()
  File "/home/exploit/.ansible/tmp/ansible-tmp-1601368297.2341213-9414-127682684873832/AnsiballZ_influxdb_database.py", line 94, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/home/exploit/.ansible/tmp/ansible-tmp-1601368297.2341213-9414-127682684873832/AnsiballZ_influxdb_database.py", line 40, in invoke_module
    runpy.run_module(mod_name='ansible_collections.community.general.plugins.modules.influxdb_database', init_globals=None, run_name='__main__', alter_sys=True)
  File "/usr/lib/python2.7/runpy.py", line 188, in run_module
    fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 82, in _run_module_code
    mod_name, mod_fname, mod_loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/tmp/ansible_influxdb_database_payload_k1Cvhm/ansible_influxdb_database_payload.zip/ansible_collections/community/general/plugins/modules/influxdb_database.py", line 141, in <module>
  File "/tmp/ansible_influxdb_database_payload_k1Cvhm/ansible_influxdb_database_payload.zip/ansible_collections/community/general/plugins/modules/influxdb_database.py", line 123, in main
  File "/tmp/ansible_influxdb_database_payload_k1Cvhm/ansible_influxdb_database_payload.zip/ansible_collections/community/general/plugins/module_utils/influxdb.py", line 88, in connect_to_influxdb
TypeError: __init__() got an unexpected keyword argument 'path'
[DEPRECATION WARNING]: Distribution Ubuntu 16.04 on host 172.20.42.11 should use /usr/bin/python3, but is using /usr/bin/python for backward compatibility with prior Ansible releases. A future Ansible release will default to using the 
discovered platform python for this host. See https://docs.ansible.com/ansible/2.10/reference_appendices/interpreter_discovery.html for more information. This feature will be removed in version 2.12. Deprecation warnings can be disabled 
by setting deprecation_warnings=False in ansible.cfg.
fatal: [172.20.42.11]: FAILED! => {
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/bin/python"
    },
    "changed": false,
    "module_stderr": "Shared connection to 172.20.42.11 closed.\r\n",
    "module_stdout": "Traceback (most recent call last):\r\n  File \"/home/exploit/.ansible/tmp/ansible-tmp-1601368297.2341213-9414-127682684873832/AnsiballZ_influxdb_database.py\", line 102, in <module>\r\n    _ansiballz_main()\r\n  File \"/home/exploit/.ansible/tmp/ansible-tmp-1601368297.2341213-9414-127682684873832/AnsiballZ_influxdb_database.py\", line 94, in _ansiballz_main\r\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\r\n  File \"/home/exploit/.ansible/tmp/ansible-tmp-1601368297.2341213-9414-127682684873832/AnsiballZ_influxdb_database.py\", line 40, in invoke_module\r\n    runpy.run_module(mod_name='ansible_collections.community.general.plugins.modules.influxdb_database', init_globals=None, run_name='__main__', alter_sys=True)\r\n  File \"/usr/lib/python2.7/runpy.py\", line 188, in run_module\r\n    fname, loader, pkg_name)\r\n  File \"/usr/lib/python2.7/runpy.py\", line 82, in _run_module_code\r\n    mod_name, mod_fname, mod_loader, pkg_name)\r\n  File \"/usr/lib/python2.7/runpy.py\", line 72, in _run_code\r\n    exec code in run_globals\r\n  File \"/tmp/ansible_influxdb_database_payload_k1Cvhm/ansible_influxdb_database_payload.zip/ansible_collections/community/general/plugins/modules/influxdb_database.py\", line 141, in <module>\r\n  File \"/tmp/ansible_influxdb_database_payload_k1Cvhm/ansible_influxdb_database_payload.zip/ansible_collections/community/general/plugins/modules/influxdb_database.py\", line 123, in main\r\n  File \"/tmp/ansible_influxdb_database_payload_k1Cvhm/ansible_influxdb_database_payload.zip/ansible_collections/community/general/plugins/module_utils/influxdb.py\", line 88, in connect_to_influxdb\r\nTypeError: __init__() got an unexpected keyword argument 'path'\r\n",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}

PLAY RECAP ************************************************************************************************************************************************************************************************************************************
172.20.42.11               : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
```

### PR DESCRIPTION
**Pull request by Kniyl**
##### SUMMARY

Make the path argument conditional on the InfluxDB version

Fixes #997 

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

influxdb_database

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by felixfontein — COMMENTED** [review]
Looks good. Maybe mention in the documentation for the `path` option that it is only used when the InfluxDB version is >= 5.1.0.

Besides that, you need a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to).

File: changelogs/fragments/1126-influxdb-conditional-path-argument.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+bugfixes:
+  - influxdb - fix usage of path for older version of
+    python-influxdb (https://github.com/ansible-collections/community.general/issues/997)
```
Comment (felixfontein) [inline changelogs/fragments/1126-influxdb-conditional-path-argument.yml:4]:
```suggestion
    python-influxdb (https://github.com/ansible-collections/community.general/issues/997).
```

File: plugins/modules/database/influxdb/influxdb_database.py [hunk 2]
```diff
@@ -16,7 +16,7 @@
 author: "Kamil Szczygiel (@kamsz)"
 requirements:
     - "python >= 2.6"
-    - "influxdb >= 0.9 & <= 1.2.4"
+    - "influxdb >= 0.9"
```
Comment (felixfontein) [inline plugins/modules/database/influxdb/influxdb_database.py:19]: Did you check whether other influxdb_* modules had the same restriction?

Comment (Kniyl) [reply plugins/modules/database/influxdb/influxdb_database.py:19]: All other influxdb modules only have the `"influxdb >= 0.9"` requirement

**Comment by Andersson007** [comment]
@resmo as an influxdb maintainer could you please take a look

**Review by resmo — APPROVED** [review]
shipit

**Pull request merged**